client/asset/dcr/dcr.go

```diff
@@ -2563,9 +2565,17 @@ func (dcr *ExchangeWallet) SwapConfirmations(ctx context.Context, coinID, contra
 	}
 	_, p2shScript := scriptAddr.PaymentScript()
 
-	// Find the contract and it's spend status using block filters.
+	// Find the contract and its spend status using block filters.
 	dcr.log.Debugf("Contract output %s:%d NOT yet found, will attempt finding it with block filters.", txHash, vout)
-	return dcr.lookupTxOutWithBlockFilters(ctx, newOutPoint(txHash, vout), p2shScript, matchTime)
+	confs, spent, err = dcr.lookupTxOutWithBlockFilters(ctx, newOutPoint(txHash, vout), p2shScript, matchTime)
+	// Don't trouble the caller if we're using an SPV wallet and the transaction
+	// cannot be located.
+	if errors.Is(err, asset.CoinNotFoundError) && dcr.wallet.SpvMode() {
+		dcr.log.Debugf("SwapConfirmations - cfilters scan did not find %v:%d. "+
+			"Assuming in mempool.", txHash, vout)
```

Comment:
I think this assumption is also valid for non-spv wallets. Is the idea that this assumption is less likely for non-spv wallets and so it's okay to alert the caller to the situation?

Reply:
:100: When this happens repeatedly with non-spv wallets, it is almost always a connectivity issue to fix promptly.

Reply:
Is it necessary to add to the comment the reason why it's okay to trouble the caller if this is not an spv wallet?